Source/WebCore/dom/IdleCallbackController.cpp

```diff
@@ -60,7 +60,8 @@ int IdleCallbackController::queueIdleCallback(Ref<IdleRequestCallback>&& callbac
                 weakThis->invokeIdleCallbackTimeout(handle);
             });
         });
-    }
+    } else if (RefPtr page = m_document ? m_document->page() : nullptr)
+        m_document->windowEventLoop().scheduleIdlePeriod(*page);
```

Comment:
`windowEventLoop()` is not inline, we need a RefPtr to the document on the stack. Also, we should likely add `Document::protectedWindowEventLoop()` and call it.

Reply:
Fixed.